arbor/backends/gpu/threshold_watcher.hpp

```diff
@@ -70,7 +70,7 @@ class threshold_watcher {
 
     /// Remove all stored crossings that were detected in previous calls to test()
     void clear_crossings() {
-        stack_.host_access();
+        stack_.refresh_host_copy();
```

Comment:
This call to `refresh_host_copy` wasn't necessary.
Now it should be 
```C++
stack_.clear();
stack_.update_device();
```

Reply:
The unit test `mc_cell_group.gpu_test` fails when I don't refresh host copy. It looks like there is a query that happens on the collected spikes after the clear. I essentially did a `get_and_clear()` call there.